test/builtin_dataset_mocks.py

```diff
@@ -482,6 +482,17 @@ def imagenet(info, root, config):
     return num_samples
 
 
+DATASET_MOCKS.append(
+    DatasetMock(
+        "imagenet",
+        mock_data_fn=imagenet_mock_data_fn,
+        configs=combinations_grid(
+            split=("train", "val", "test"),
+        ),
```

Comment:
Kind of related to my comment above about `ADDITIONAL_CONFIGS`, possibly as an alternative:

Would it be reasonably feasible  to have a `@register_mock` decorator to which we would pass the `configs`? Something like

```py
@register_mock(configs=combinations_grid(...))
def imagenet_mock_data_fn(...):
     ....
```


```

Reply:
That works here, yes. If the `mock_data_fn` is the only function that we need for a dataset mock, I think this is preferable to overwriting `DatasetMock`.

Reply:
OK sounds good.
TBH it's also perfectly fine to keep what we currently have `DATASET_MOCKS.append(...)` as a first version. I might be early-optimizing here.